pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
`ignoreRepository` is true if the `SourceRepository` was created by either `NewSourceRepositoryForDockerfile` or `NewBinarySourceRepository`.

Unless we'll dig deeper and make further tests, I don't want to change that logic.

The current patch makes it be "whatever it was before + make `oc new-build <repo> --dockerfile` work".